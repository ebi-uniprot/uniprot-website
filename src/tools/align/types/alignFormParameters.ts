import { ServerParameters } from './alignServerParameters';

export type FormParameters = {
  sequence: ServerParameters['sequence'];
};
