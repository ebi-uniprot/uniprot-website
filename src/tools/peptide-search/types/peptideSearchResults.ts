/* Results as given by the server */

export type PeptideSearchResults = string;
