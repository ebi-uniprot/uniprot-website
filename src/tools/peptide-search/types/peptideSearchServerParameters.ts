/* Parameters of a peptide search job as required by the server */
// http://peptidesearch.uniprot.org/

export type PepS = string;

export type TaxIds = string;

export type LEQi = 'on' | 'off';

export type SpOnly = 'on' | 'off';

export type ServerParameters = {
  peps: PepS;
  taxIds?: TaxIds;
  lEQi: LEQi;
  spOnly: SpOnly;
};

// same, because no email was provided before
export type PublicServerParameters = ServerParameters;
