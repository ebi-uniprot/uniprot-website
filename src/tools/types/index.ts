export enum Tool {
  blast = 'blast',
  // align = 'align',
  // peptideSearch = 'peptideSearch',
  // idMapping = 'idMapping',
}
