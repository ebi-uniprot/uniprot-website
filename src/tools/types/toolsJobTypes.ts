export enum JobTypes {
  BLAST = 'blast',
  ALIGN = 'align',
  UPLOAD_LIST = 'upload list',
  PEPTIDE_SEARCH = 'peptide search',
}
