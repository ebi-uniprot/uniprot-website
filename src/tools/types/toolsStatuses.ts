/* Job statuses as given by the server */

// as returned by https://www.ebi.ac.uk/Tools/services/rest/ncbiblast/status/{jobId}
// as returned by https://www.ebi.ac.uk/Tools/services/rest/clustalo/status/{jobId}
export enum Status {
  NEW = 'NEW', // Specific to async-download: indicates created and queued. Id mapping does use this but we don't see it because it's never queued.
  CREATED = 'CREATED', // not from the server, for internal use
  ERRORED = 'ERRORED', // according to www-prod, for problems with the tool
  FAILURE = 'FAILURE', // according to www-prod, when something else in the pipeline failed (LSF issues, etc.)
  RUNNING = 'RUNNING',
  FINISHED = 'FINISHED',
  NOT_FOUND = 'NOT_FOUND',
}
