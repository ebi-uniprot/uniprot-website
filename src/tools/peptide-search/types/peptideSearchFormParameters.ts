import { PepS, LEQi, SpOnly } from './peptideSearchServerParameters';
import { SelectedTaxon } from '../../types/toolsFormData';

export type FormParameters = {
  peps: PepS;
  taxIds?: SelectedTaxon[];
  lEQi: LEQi;
  spOnly: SpOnly;
};
