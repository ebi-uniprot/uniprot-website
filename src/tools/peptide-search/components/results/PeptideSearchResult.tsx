import { useMemo, useRef, lazy, Suspense } from 'react';
import { Link } from 'react-router-dom';
import {
  Loader,
  LongNumber,
  Message,
  PageIntro,
  Tab,
  Tabs,
} from 'franklin-sites';
import { partialRight } from 'lodash-es';

import useDataApi from '../../../../shared/hooks/useDataApi';
import useDataApiWithStale from '../../../../shared/hooks/useDataApiWithStale';
import useNSQuery from '../../../../shared/hooks/useNSQuery';
import useMarkJobAsSeen from '../../../hooks/useMarkJobAsSeen';
import { useToolsState } from '../../../../shared/contexts/Tools';
import useMatchWithRedirect from '../../../../shared/hooks/useMatchWithRedirect';
import usePaginatedAccessions from '../../../../shared/hooks/usePaginatedAccessions';

import HTMLHead from '../../../../shared/components/HTMLHead';
import ErrorBoundary from '../../../../shared/components/error-component/ErrorBoundary';
import ResultsFacets from '../../../../shared/components/results/ResultsFacets';
import { SidebarLayout } from '../../../../shared/components/layouts/SideBarLayouttest';
import NoResultsPage from '../../../../shared/components/error-pages/NoResultsPage';
import ErrorHandler from '../../../../shared/components/error-pages/ErrorHandler';

import toolsURLs from '../../../config/urls';
import {
  Namespace,
  namespaceAndToolsLabels,
} from '../../../../shared/types/namespaces';
import { Location, changePathnameOnly } from '../../../../app/config/urls';
import peptideSearchConverter from '../../adapters/peptideSearchConverter';

import { UniProtkbAPIModel } from '../../../../uniprotkb/adapters/uniProtkbConverter';
import { SearchResults } from '../../../../shared/types/results';
import { PeptideSearchResults } from '../../types/peptideSearchResults';
import { JobTypes } from '../../../types/toolsJobTypes';
import { Status } from '../../../types/toolsStatuses';
import { FinishedJob } from '../../../types/toolsJob';
import { ToolsState } from '../../../state/toolsInitialState';

import helper from '../../../../shared/styles/helper.module.scss';
import sidebarStyles from '../../../../shared/components/layouts/styles/sidebar-layout.module.scss';

const jobType = JobTypes.PEPTIDE_SEARCH;
const urls = toolsURLs(jobType);
const title = `${namespaceAndToolsLabels[jobType]} results`;

// overview
const PeptideSearchResultTable = lazy(
  () =>
    import(
      /* webpackChunkName: "peptide-search-result-page" */ './PeptideSearchResultTable'
    )
);
// input-parameters
const InputParameters = lazy(
  () =>
    import(
      /* webpackChunkName: "input-parameters" */ '../../../components/InputParameters'
    )
);
// input-parameters
const APIRequest = lazy(
  () =>
    import(
      /* webpackChunkName: "api-request" */ '../../../components/APIRequest'
    )
);

enum TabLocation {
  Overview = 'overview',
  InputParameters = 'input-parameters',
  APIRequest = 'api-request',
}

type Params = {
  id: string;
  subPage?: TabLocation;
};

const MAX_FACETS = 1_000;

const PeptideSearchResult = ({
  toolsState,
}: {
  toolsState: NonNullable<ToolsState>;
}) => {
  const match = useMatchWithRedirect<Params>(
    Location.PeptideSearchResult,
    TabLocation
  );

  const jobSubmission = useRef<FinishedJob<JobTypes.PEPTIDE_SEARCH> | null>(
    null
  );
  const jobID = match?.params.id || '';
  const {
    data: jobResultData,
    loading: jobResultLoading,
    error: jobResultError,
    status: jobResultStatus,
  } = useDataApi<PeptideSearchResults>(urls.resultUrl(jobID, {}), {
    headers: { accept: 'text/plain' },
  });

  // Get the job mission from local tools state. Save this in a reference as
  // the job may change with useMarkJobAsSeen but we don't want to have to
  // recreate the converter which will cause usePagination to duplicate rows
  if (!jobSubmission.current) {
    jobSubmission.current =
      Object.values(toolsState).find(
        (
          jobSubmission
        ): jobSubmission is FinishedJob<JobTypes.PEPTIDE_SEARCH> =>
          jobSubmission.status === Status.FINISHED &&
          jobSubmission?.remoteID === jobID
      ) || null;
  }

  const accessions = useMemo(
    () => jobResultData?.split(',').filter(Boolean),
    [jobResultData]
  );

  const excessAccessions = accessions && accessions?.length > MAX_FACETS;

  // Query for facets
  const initialApiFacetUrl = useNSQuery({
    size: 0,
    withFacets: true,
    withColumns: false,
    accessions: excessAccessions ? [] : accessions,
  });
  const facetApiObject =
    useDataApiWithStale<SearchResults<UniProtkbAPIModel>>(initialApiFacetUrl);
  const {
    loading: facetInititialLoading,
    headers: facetHeaders,
    isStale: facetHasStaleData,
  } = facetApiObject;
  const facetTotal = facetHeaders?.['x-total-results'];

  const converter = useMemo(() => {
    const peps = jobSubmission.current?.parameters.peps;
    return peps ? partialRight(peptideSearchConverter, peps) : undefined;
  }, [jobSubmission]);

  // TODO: if the user didn't submit this jobSubmission there is no way to get the initial sequence. In the
  // future the API may provide this information in which case we would want to fetch and show
  const resultsDataObject = usePaginatedAccessions(accessions, converter);

  const {
    initialLoading: resultsDataInitialLoading,
    total: resultsDataTotal,
    progress: resultsDataProgress,
  } = resultsDataObject;

  useMarkJobAsSeen(resultsDataObject?.allResults.length, match?.params.id);

  let total: undefined | number;
  if (facetTotal !== undefined) {
    total = +facetTotal;
  }
  if (resultsDataTotal !== undefined) {
    total = +resultsDataTotal;
  }

  if (jobResultError || !match) {
    return <ErrorHandler status={jobResultStatus} />;
  }

  if (
    (!jobResultLoading &&
      !resultsDataInitialLoading &&
      !facetInititialLoading &&
      !total &&
      !excessAccessions) ||
    (!jobResultLoading && accessions?.length === 0) ||
    total === 0
  ) {
    return <NoResultsPage />;
  }

  if (
    jobResultLoading ||
    (facetInititialLoading && resultsDataInitialLoading && !facetHasStaleData)
  ) {
    return <Loader progress={resultsDataProgress} />;
  }

  let sidebar: JSX.Element;
  // Deciding what should be displayed on the sidebar
  switch (match.params.subPage) {
    case TabLocation.InputParameters:
    case TabLocation.APIRequest:
      sidebar = <div className={sidebarStyles['empty-sidebar']} />;
      break;

    default:
      sidebar = (
        <ErrorBoundary>
          <ResultsFacets
            dataApiObject={facetApiObject}
            namespaceOverride={Namespace.uniprotkb}
          />
        </ErrorBoundary>
      );
      break;
  }

  if (excessAccessions) {
    sidebar = <div className={sidebarStyles['empty-sidebar']} />;
  }
  const basePath = `/peptide-search/${match.params.id}/`;

  return (
    <SidebarLayout sidebar={sidebar}>
      <HTMLHead title={title}>
        <meta name="robots" content="noindex" />
      </HTMLHead>
      <PageIntro
        title={namespaceAndToolsLabels[JobTypes.PEPTIDE_SEARCH]}
        titlePostscript={
          total && (
            <small>
              found in {namespaceAndToolsLabels[Namespace.uniprotkb]}
            </small>
          )
        }
        resultsCount={total}
      />
      <Tabs
        active={match.params.subPage}
        className={jobResultLoading ? helper.stale : undefined}
      >
        <Tab
          id={TabLocation.Overview}
          title={
            <Link to={changePathnameOnly(basePath + TabLocation.Overview)}>
              Overview
            </Link>
          }
        >
          <Suspense fallback={<Loader />}>
            {excessAccessions && (
              <Message level="warning">
                Filters are not supported for peptide results if there are more
                than <LongNumber>{MAX_FACETS}</LongNumber> matches.
              </Message>
            )}

            <PeptideSearchResultTable
              total={total}
              resultsDataObject={resultsDataObject}
              accessions={accessions}
              jobSubmission={jobSubmission}
            />
          </Suspense>
        </Tab>
        <Tab
          id={TabLocation.InputParameters}
          title={
            <Link
              to={changePathnameOnly(basePath + TabLocation.InputParameters)}
            >
              Input Parameters
            </Link>
          }
        >
          <HTMLHead title={[title, 'Input Parameters']} />
          <Suspense fallback={<Loader />}>
            <InputParameters
              id={match.params.id}
              inputParamsData={jobSubmission.current?.parameters}
              jobType={jobType}
            />
          </Suspense>
        </Tab>
        <Tab
          id={TabLocation.APIRequest}
          title={
            <Link to={changePathnameOnly(basePath + TabLocation.APIRequest)}>
              API Request
            </Link>
          }
        >
          <HTMLHead title={[title, 'API Request']} />
          <Suspense fallback={<Loader />}>
            <APIRequest
              jobType={jobType}
              inputParamsData={jobSubmission.current?.parameters}
            />
          </Suspense>
        </Tab>
      </Tabs>
    </SidebarLayout>
  );
};

const PeptideSearchResultWithToolsState = () => {
  const toolsState = useToolsState();
  return toolsState === null ? (
    <Loader />
  ) : (
    <PeptideSearchResult toolsState={toolsState} />
  );
};

export default PeptideSearchResultWithToolsState;
