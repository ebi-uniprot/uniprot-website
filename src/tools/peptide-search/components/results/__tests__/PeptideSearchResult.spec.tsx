import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { screen } from '@testing-library/react';

import customRender from '../../../../../shared/__test-helpers__/customRender';

import PeptideSearchResult from '../PeptideSearchResult';

import { FinishedJob } from '../../../../types/toolsJob';
import { JobTypes } from '../../../../types/toolsJobTypes';
import { Status } from '../../../../types/toolsStatuses';

import uniprotkbResults from '../../../../../uniprotkb/components/__mocks__/results';

const mockJob: FinishedJob<JobTypes.PEPTIDE_SEARCH> = {
  internalID: 'local-id',
  parameters: {
    peps: 'AAQGYGYYRTVIFSAMFGGYSLYYFNRKTFSF\nIQSTHYLQVNYQDSQDWFILVSVIADLRNAFYVLFPIWFHLQEAVGI',
    taxIds: undefined,
    lEQi: 'off',
    spOnly: 'on',
  },
  remoteID: 'remote-id',
  saved: false,
  status: Status.FINISHED,
  timeCreated: 1620215833639,
  timeFinished: 1620216241725,
  timeLastUpdate: 1620216241725,
  timeSubmitted: 1620215833848,
  seen: false,
  title: 'title',
  type: JobTypes.PEPTIDE_SEARCH,
};

const mockRequests = new MockAdapter(axios);
mockRequests
  .onGet(new RegExp(`/peptidematchws/asyncrest/jobs/${mockJob.remoteID}$`))
  .reply(200, 'P35575,O43826');

mockRequests
  .onGet(/uniprotkb\/accessions/)
  .reply(
    200,
    { results: uniprotkbResults.results.slice(0, 2) },
    { 'x-total-results': 2 }
  );

describe('PeptideSearchResult', () => {
  it('should render with the correct number of results in the title, not own job', async () => {
    const { asFragment } = customRender(<PeptideSearchResult />, {
      route: `/peptide-search/${mockJob.remoteID}`,
      initialLocalStorage: {
        'view-mode': 'table',
      },
    });
    await screen.findByText('2 results');
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with the correct number of results in the title, own user job', async () => {
    const { asFragment } = customRender(<PeptideSearchResult />, {
      route: `/peptide-search/${mockJob.remoteID}`,
      initialLocalStorage: {
        'view-mode': 'table',
      },
      toolsState: { [mockJob.internalID]: mockJob },
    });
    await screen.findByText('2 results');
    expect(asFragment()).toMatchSnapshot();
  });
});
