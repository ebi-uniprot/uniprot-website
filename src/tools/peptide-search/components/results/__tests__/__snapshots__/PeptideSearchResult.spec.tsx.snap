// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeptideSearchResult should render with the correct number of results in the title, not own job 1`] = `
<DocumentFragment>
  <div
    class="sidebar-layout"
  >
    <section
      class="sidebar-layout__sidebar"
    />
    <section
      class="sidebar-layout__content"
    >
      <section
        class="sidebar-layout__title"
      >
        <div
          class="page-intro"
        >
          <h1
            class=""
          >
            Peptide search
            <small>
               2 results 
            </small>
            <small>
              in UniProtKB
            </small>
          </h1>
        </div>
      </section>
      <div
        class="tabs"
      >
        <div
          class="tabs__header"
          role="tablist"
        >
          <div
            aria-controls="1"
            class="tabs__header__item tabs__header__item--active"
            data-target="overview"
            data-testid="tab-title"
            role="tab"
          >
            <a
              href="/peptide-search/remote-id/overview"
            >
              Overview
            </a>
          </div>
          <div
            aria-controls="1"
            class="tabs__header__item"
            data-target="input-parameters"
            data-testid="tab-title"
            role="tab"
          >
            <a
              href="/peptide-search/remote-id/input-parameters"
            >
              Input Parameters
            </a>
          </div>
          <div
            aria-controls="1"
            class="tabs__header__item"
            data-target="api-request"
            data-testid="tab-title"
            role="tab"
          >
            <a
              href="/peptide-search/remote-id/api-request"
            >
              API Request
            </a>
          </div>
        </div>
        <div
          data-testid="tab-content"
          id="1"
          role="tabpanel"
        >
          <div
            style="display: block;"
          >
            <div
              class="button-group results-buttons"
            >
              <button
                class="button tertiary disabled"
                disabled=""
                title="Select at least one entry to run a BLAST job"
                type="button"
              >
                BLAST
              </button>
              <button
                class="button tertiary disabled"
                disabled=""
                title="Select at least 2 entries to run an Align job"
                type="button"
              >
                Align
              </button>
              <button
                class="button tertiary disabled"
                disabled=""
                title="Select at least one entry to map"
                type="button"
              >
                Map IDs
              </button>
              <button
                class="button tertiary"
                type="button"
              >
                <test-file-stub />
                Download
              </button>
              <button
                class="button tertiary disabled"
                disabled=""
                title="Select at least one entry to add to the basket"
                type="button"
              >
                <test-file-stub />
                Add
              </button>
              <span
                role="radiogroup"
              >
                View:
                <label>
                  Table 
                  <input
                    checked=""
                    name="view"
                    type="radio"
                  />
                </label>
                <label>
                  Card 
                  <input
                    name="view"
                    type="radio"
                  />
                </label>
              </span>
              <button
                class="button tertiary"
                type="button"
              >
                <test-file-stub />
                Customize columns
              </button>
            </div>
            <div
              class="results-data"
            >
              <table
                class="data-table"
              >
                <thead>
                  <tr>
                    <th
                      class="data-table__header-cell--checkbox"
                    >
                      <div
                        class="checkbox-cell"
                      >
                        <input
                          id="3"
                          type="checkbox"
                        />
                        <label
                          aria-label="Selection control for all visible items"
                          for="3"
                        />
                      </div>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        aria-expanded="false"
                      >
                        <span>
                          Entry
                        </span>
                      </span>
                    </th>
                    <th
                      class=""
                    />
                    <th
                      class=""
                    >
                      <span
                        aria-expanded="false"
                      >
                        <span
                          data-article-id="entry_name"
                        >
                          Entry Name
                        </span>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        aria-expanded="false"
                      >
                        <span
                          data-article-id="protein_names"
                        >
                          Protein Names
                        </span>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        aria-expanded="false"
                      >
                        <span
                          data-article-id="gene_name"
                        >
                          Gene Names
                        </span>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        aria-expanded="false"
                      >
                        <span
                          data-article-id="organism-name"
                        >
                          Organism
                        </span>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        aria-expanded="false"
                      >
                        <span>
                          Length
                        </span>
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td
                      class="checkbox-cell"
                    >
                      <input
                        data-id="P35575"
                        id="6"
                        type="checkbox"
                      />
                      <label
                        aria-label="P35575"
                        for="6"
                        title="click to select, shift+click for multiple selection"
                      />
                    </td>
                    <td>
                      <span
                        class="no-wrap accession-view"
                      >
                        <a
                          class="accession"
                          href="/uniprotkb/P35575"
                        >
                          P35575
                        </a>
                      </span>
                    </td>
                    <td>
                      <span
                        class="entry-title__status icon--reviewed"
                        title="This marks a reviewed UniProtKB entry"
                      >
                        <test-file-stub />
                      </span>
                    </td>
                    <td>
                      <span>
                        G6PC1_HUMAN
                      </span>
                    </td>
                    <td>
                      <span
                        style="font-weight: bolder;"
                      >
                        Glucose-6-phosphatase catalytic subunit 1
                      </span>
                      <button
                        aria-expanded="false"
                        class="button tertiary ellipsis-reveal"
                        title="Show more"
                        type="button"
                      >
                        [...]
                      </button>
                    </td>
                    <td>
                      <span
                        style="font-weight: bolder;"
                      >
                        G6PC1
                      </span>
                      <button
                        aria-expanded="false"
                        class="button tertiary ellipsis-reveal"
                        title="Show more"
                        type="button"
                      >
                        [...]
                      </button>
                    </td>
                    <td>
                      <a
                        href="/taxonomy/9606"
                        title="Homo sapiens (Human), taxon ID 9606"
                      >
                        Homo sapiens (Human)
                      </a>
                    </td>
                    <td>
                      357 AA
                    </td>
                  </tr>
                  <tr>
                    <td
                      class="checkbox-cell"
                    >
                      <input
                        data-id="O43826"
                        id="7"
                        type="checkbox"
                      />
                      <label
                        aria-label="O43826"
                        for="7"
                        title="click to select, shift+click for multiple selection"
                      />
                    </td>
                    <td>
                      <span
                        class="no-wrap accession-view"
                      >
                        <a
                          class="accession"
                          href="/uniprotkb/O43826"
                        >
                          O43826
                        </a>
                      </span>
                    </td>
                    <td>
                      <span
                        class="entry-title__status icon--reviewed"
                        title="This marks a reviewed UniProtKB entry"
                      >
                        <test-file-stub />
                      </span>
                    </td>
                    <td>
                      <span>
                        G6PT1_HUMAN
                      </span>
                    </td>
                    <td>
                      <span
                        style="font-weight: bolder;"
                      >
                        Glucose-6-phosphate exchanger SLC37A4
                      </span>
                      <button
                        aria-expanded="false"
                        class="button tertiary ellipsis-reveal"
                        title="Show more"
                        type="button"
                      >
                        [...]
                      </button>
                    </td>
                    <td>
                      <span
                        style="font-weight: bolder;"
                      >
                        SLC37A4
                      </span>
                      <button
                        aria-expanded="false"
                        class="button tertiary ellipsis-reveal"
                        title="Show more"
                        type="button"
                      >
                        [...]
                      </button>
                    </td>
                    <td>
                      <a
                        href="/taxonomy/9606"
                        title="Homo sapiens (Human), taxon ID 9606"
                      >
                        Homo sapiens (Human)
                      </a>
                    </td>
                    <td>
                      429 AA
                    </td>
                  </tr>
                </tbody>
              </table>
              <div
                class="data-loader__loading"
              />
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`PeptideSearchResult should render with the correct number of results in the title, own user job 1`] = `
<DocumentFragment>
  <div
    class="sidebar-layout"
  >
    <section
      class="sidebar-layout__sidebar"
    />
    <section
      class="sidebar-layout__content"
    >
      <section
        class="sidebar-layout__title"
      >
        <div
          class="page-intro"
        >
          <h1
            class=""
          >
            Peptide search
            <small>
               2 results 
            </small>
            <small>
              in UniProtKB
            </small>
          </h1>
        </div>
      </section>
      <div
        class="tabs"
      >
        <div
          class="tabs__header"
          role="tablist"
        >
          <div
            aria-controls="2"
            class="tabs__header__item tabs__header__item--active"
            data-target="overview"
            data-testid="tab-title"
            role="tab"
          >
            <a
              href="/peptide-search/remote-id/overview"
            >
              Overview
            </a>
          </div>
          <div
            aria-controls="2"
            class="tabs__header__item"
            data-target="input-parameters"
            data-testid="tab-title"
            role="tab"
          >
            <a
              href="/peptide-search/remote-id/input-parameters"
            >
              Input Parameters
            </a>
          </div>
          <div
            aria-controls="2"
            class="tabs__header__item"
            data-target="api-request"
            data-testid="tab-title"
            role="tab"
          >
            <a
              href="/peptide-search/remote-id/api-request"
            >
              API Request
            </a>
          </div>
        </div>
        <div
          data-testid="tab-content"
          id="2"
          role="tabpanel"
        >
          <div
            style="display: block;"
          >
            <div
              class="button-group results-buttons"
            >
              <button
                class="button tertiary disabled"
                disabled=""
                title="Select at least one entry to run a BLAST job"
                type="button"
              >
                BLAST
              </button>
              <button
                class="button tertiary disabled"
                disabled=""
                title="Select at least 2 entries to run an Align job"
                type="button"
              >
                Align
              </button>
              <button
                class="button tertiary disabled"
                disabled=""
                title="Select at least one entry to map"
                type="button"
              >
                Map IDs
              </button>
              <button
                class="button tertiary"
                type="button"
              >
                <test-file-stub />
                Download
              </button>
              <button
                class="button tertiary disabled"
                disabled=""
                title="Select at least one entry to add to the basket"
                type="button"
              >
                <test-file-stub />
                Add
              </button>
              <span
                role="radiogroup"
              >
                View:
                <label>
                  Table 
                  <input
                    checked=""
                    name="view"
                    type="radio"
                  />
                </label>
                <label>
                  Card 
                  <input
                    name="view"
                    type="radio"
                  />
                </label>
              </span>
              <button
                class="button tertiary"
                type="button"
              >
                <test-file-stub />
                Customize columns
              </button>
            </div>
            <div
              class="results-data"
            >
              <table
                class="data-table"
              >
                <thead>
                  <tr>
                    <th
                      class="data-table__header-cell--checkbox"
                    >
                      <div
                        class="checkbox-cell"
                      >
                        <input
                          id="4"
                          type="checkbox"
                        />
                        <label
                          aria-label="Selection control for all visible items"
                          for="4"
                        />
                      </div>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        aria-expanded="false"
                      >
                        <span>
                          Entry
                        </span>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        aria-expanded="false"
                      >
                        <span>
                          Match
                        </span>
                      </span>
                    </th>
                    <th
                      class=""
                    />
                    <th
                      class=""
                    >
                      <span
                        aria-expanded="false"
                      >
                        <span
                          data-article-id="entry_name"
                        >
                          Entry Name
                        </span>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        aria-expanded="false"
                      >
                        <span
                          data-article-id="protein_names"
                        >
                          Protein Names
                        </span>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        aria-expanded="false"
                      >
                        <span
                          data-article-id="gene_name"
                        >
                          Gene Names
                        </span>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        aria-expanded="false"
                      >
                        <span
                          data-article-id="organism-name"
                        >
                          Organism
                        </span>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span
                        aria-expanded="false"
                      >
                        <span>
                          Length
                        </span>
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td
                      class="checkbox-cell"
                    >
                      <input
                        data-id="P35575"
                        id="7"
                        type="checkbox"
                      />
                      <label
                        aria-label="P35575"
                        for="7"
                        title="click to select, shift+click for multiple selection"
                      />
                    </td>
                    <td>
                      <span
                        class="no-wrap accession-view"
                      >
                        <a
                          class="accession"
                          href="/uniprotkb/P35575"
                        >
                          P35575
                        </a>
                      </span>
                    </td>
                    <td>
                      <ul
                        class="no-bullet no-wrap"
                      >
                        <li>
                          Positions 13-59: IQSTHYLQVN
                          <button
                            aria-expanded="false"
                            class="button tertiary ellipsis-reveal"
                            title="Show more"
                            type="button"
                          >
                            [...]
                          </button>
                        </li>
                      </ul>
                    </td>
                    <td>
                      <span
                        class="entry-title__status icon--reviewed"
                        title="This marks a reviewed UniProtKB entry"
                      >
                        <test-file-stub />
                      </span>
                    </td>
                    <td>
                      <span>
                        G6PC1_HUMAN
                      </span>
                    </td>
                    <td>
                      <span
                        style="font-weight: bolder;"
                      >
                        Glucose-6-phosphatase catalytic subunit 1
                      </span>
                      <button
                        aria-expanded="false"
                        class="button tertiary ellipsis-reveal"
                        title="Show more"
                        type="button"
                      >
                        [...]
                      </button>
                    </td>
                    <td>
                      <span
                        style="font-weight: bolder;"
                      >
                        G6PC1
                      </span>
                      <button
                        aria-expanded="false"
                        class="button tertiary ellipsis-reveal"
                        title="Show more"
                        type="button"
                      >
                        [...]
                      </button>
                    </td>
                    <td>
                      <a
                        href="/taxonomy/9606"
                        title="Homo sapiens (Human), taxon ID 9606"
                      >
                        Homo sapiens (Human)
                      </a>
                    </td>
                    <td>
                      357 AA
                    </td>
                  </tr>
                  <tr>
                    <td
                      class="checkbox-cell"
                    >
                      <input
                        data-id="O43826"
                        id="8"
                        type="checkbox"
                      />
                      <label
                        aria-label="O43826"
                        for="8"
                        title="click to select, shift+click for multiple selection"
                      />
                    </td>
                    <td>
                      <span
                        class="no-wrap accession-view"
                      >
                        <a
                          class="accession"
                          href="/uniprotkb/O43826"
                        >
                          O43826
                        </a>
                      </span>
                    </td>
                    <td>
                      <ul
                        class="no-bullet no-wrap"
                      >
                        <li>
                          Positions 2-33: AAQGYGYYRT
                          <button
                            aria-expanded="false"
                            class="button tertiary ellipsis-reveal"
                            title="Show more"
                            type="button"
                          >
                            [...]
                          </button>
                        </li>
                      </ul>
                    </td>
                    <td>
                      <span
                        class="entry-title__status icon--reviewed"
                        title="This marks a reviewed UniProtKB entry"
                      >
                        <test-file-stub />
                      </span>
                    </td>
                    <td>
                      <span>
                        G6PT1_HUMAN
                      </span>
                    </td>
                    <td>
                      <span
                        style="font-weight: bolder;"
                      >
                        Glucose-6-phosphate exchanger SLC37A4
                      </span>
                      <button
                        aria-expanded="false"
                        class="button tertiary ellipsis-reveal"
                        title="Show more"
                        type="button"
                      >
                        [...]
                      </button>
                    </td>
                    <td>
                      <span
                        style="font-weight: bolder;"
                      >
                        SLC37A4
                      </span>
                      <button
                        aria-expanded="false"
                        class="button tertiary ellipsis-reveal"
                        title="Show more"
                        type="button"
                      >
                        [...]
                      </button>
                    </td>
                    <td>
                      <a
                        href="/taxonomy/9606"
                        title="Homo sapiens (Human), taxon ID 9606"
                      >
                        Homo sapiens (Human)
                      </a>
                    </td>
                    <td>
                      429 AA
                    </td>
                  </tr>
                </tbody>
              </table>
              <div
                class="data-loader__loading"
              />
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
